opentech/apply/projects/forms.py

```diff
@@ -54,32 +71,40 @@ def clean(self):
         super().clean()
 
 
-class ChangePaymentRequestStatusForm(forms.ModelForm):
-    name = 'change_payment_request_status_form'
+class ChangePaymentRequestStatusForm(LoopedFormMixin, forms.ModelForm):
+    name_prefix = 'change_payment_request_status_form'
 
     class Meta:
-        fields = ['status']
+        fields = ['status', 'comment', 'paid_value']
         model = PaymentRequest
 
     def __init__(self, instance, *args, **kwargs):
         super().__init__(instance=instance, *args, **kwargs)
 
         self.instance = instance
+        self.initial['paid_value'] = instance.requested_value
 
         status_field = self.fields['status']
 
-        if instance.status == SUBMITTED:
-            wanted = [CHANGES_REQUESTED, UNDER_REVIEW, PAID, DECLINED]
-            status_field.choices = filter_request_choices(wanted)
-            return
+        statuses_lut = {
+            CHANGES_REQUESTED: filter_request_choices([DECLINED]),
+            SUBMITTED: filter_request_choices([CHANGES_REQUESTED, UNDER_REVIEW, PAID, DECLINED]),
+            UNDER_REVIEW: filter_request_choices([PAID]),
+        }
+        status_field.choices = statuses_lut.get(instance.status, status_field.choices)
```

Comment:
Whats the scenario where instance.status is not in `statuses_lut`. I feel returning all choices in that situation might be less safe than returning no choices or a list of "safe" choices.

Reply:
When `instance.status` is either `PAID` or `DECLINED`.  There are checks elsewhere to stop the user reaching that state but I could return an empty list here too?

Reply:
Please,

Reply:
Fixed and rebased.